hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/handlers/NodeCreateHandler.java

```diff
@@ -0,0 +1,122 @@
+import edu.umd.cs.findbugs.annotations.NonNull;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+/**
+ * This class contains all workflow-related functionality regarding {@link HederaFunctionality#NODE_CREATE}.
+ */
+@Singleton
+public class NodeCreateHandler implements TransactionHandler {
+    private final AddressBookValidator addressBookValidator;
+
+    /**
+     * Constructs a {@link NodeCreateHandler} with the given {@link AddressBookValidator}.
+     * @param addressBookValidator the validator for the crypto create transaction
+     */
+    @Inject
+    public NodeCreateHandler(@NonNull final AddressBookValidator addressBookValidator) {
+        this.addressBookValidator =
+                requireNonNull(addressBookValidator, "The supplied argument 'addressBookValidator' must not be null");
+    }
+
+    @Override
+    public void pureChecks(@NonNull final TransactionBody txn) throws PreCheckException {
+        final var op = txn.nodeCreate();
+        validateTruePreCheck(op != null, INVALID_TRANSACTION_BODY);
+        final var accountId = op.accountIdOrElse(AccountID.DEFAULT);
+        validateFalsePreCheck(accountId.equals(AccountID.DEFAULT), INVALID_NODE_ACCOUNT_ID);
+        validateFalsePreCheck(!accountId.hasAccountNum() && accountId.hasAlias(), INVALID_NODE_ACCOUNT_ID);
+        validateFalsePreCheck(op.gossipEndpoint().isEmpty(), INVALID_GOSSIP_ENDPOINT);
+        validateFalsePreCheck(op.serviceEndpoint().isEmpty(), INVALID_SERVICE_ENDPOINT);
+        validateFalsePreCheck(op.gossipCaCertificate().length() == 0, INVALID_GOSSIP_CAE_CERTIFICATE);
+    }
+
+    @Override
+    public void preHandle(@NonNull final PreHandleContext context) throws PreCheckException {
+        requireNonNull(context);
+        final var op = context.body().nodeCreate();
+        final var accountStore = context.createStore(ReadableAccountStore.class);
+        final var accountId = op.accountIdOrElse(AccountID.DEFAULT);
+
+        validateTruePreCheck(accountStore.contains(accountId), INVALID_NODE_ACCOUNT_ID);
+    }
+
+    @Override
+    public void handle(@NonNull final HandleContext handleContext) {
+        requireNonNull(handleContext);
+
+        final var op = handleContext.body().nodeCreate();
+
+        final var configuration = handleContext.configuration();
+        final var nodeConfig = configuration.getConfigData(NodesConfig.class);
+        final var nodeStore = handleContext.writableStore(WritableNodeStore.class);
+
+        final var nodeBuilder = new Node.Builder();
+
+        validateFalse(nodeStore.sizeOfState() >= nodeConfig.maxNumber(), MAX_NODE_HAVE_BEEN_CREATED);
+        addressBookValidator.validateDescription(op.description(), nodeConfig);
+        addressBookValidator.validateGossipEndpoint(op.gossipEndpoint(), nodeConfig);
+        addressBookValidator.validateServiceEndpoint(op.serviceEndpoint(), nodeConfig);
+        validateFalse(op.gossipCaCertificate().length() == 0, INVALID_GOSSIP_CAE_CERTIFICATE);
+
```

Comment:
This should be in `pureChecks`

Reply:
I had it in pureChecks(), forgot remove from handle(). Removed it.